src/lib.rs

```diff
@@ -176,6 +176,62 @@ impl<F: Float, C: FloatChecker<F>> NoisyFloat<F, C> {
         }
     }
 
+    /// Converts the value in-place to a reference to a `NoisyFloat`.
+    ///
+    /// Uses the `FloatChecker` to assert that the value is valid.
+    #[inline]
+    pub fn borrowed(value: &F) -> &Self {
+        C::assert(*value);
+        Self::unchecked_borrowed(value)
+    }
+
+    #[inline]
+    fn unchecked_borrowed(value: &F) -> &Self {
+        // This is safe because `NoisyFloat` is a thin wrapper around the
+        // floating-point type.
+        unsafe { &*(value as *const F as *const Self) }
+    }
+
+    /// Tries to convert the value in-place to a reference to a `NoisyFloat`.
+    ///
+    /// Returns `None` if the value is invalid.
+    #[inline]
+    pub fn try_borrowed(value: &F) -> Option<&Self> {
+        if C::check(*value) {
+            Some(Self::unchecked_borrowed(value))
+        } else {
+            None
+        }
+    }
+
+    /// Converts the value in-place to a mutable reference to a `NoisyFloat`.
+    ///
+    /// Uses the `FloatChecker` to assert that the value is valid.
+    #[inline]
+    pub fn borrowed_mut(value: &mut F) -> &mut Self {
```

Comment:
Since borrowed_mut(...) relies on a tricky unsafe cast, you should add a unit test to make sure it works as intended.

Reply:
See above.